api/src/main/java/org/openmrs/module/rwandaemr/radiology/OrderToORMTranslator.java

```diff
@@ -0,0 +1,220 @@
+
+    private final Parser parser = new PipeParser();
+
+    private final AdtService adtService;
+    private final ConceptService conceptService;
+    private final RwandaEmrConfig rwandaEmrConfig;
+
+    public OrderToORMTranslator(
+            @Autowired AdtService adtService,
+            @Autowired ConceptService conceptService,
+            @Autowired RwandaEmrConfig rwandaEmrConfig) {
+        this.adtService = adtService;
+        this.conceptService = conceptService;
+        this.rwandaEmrConfig = rwandaEmrConfig;
+    }
+
+    /**
+     * For the given test order, generate an ORM^001 HL7 message
+     * Used for a new unscheduled order creation or order cancellation
+     */
+    public String toORM_001(TestOrder order) throws HL7Exception {
+        ORM_O01 message = new ORM_O01();
+        Date now = new Date();
+        DateFormat dateFormat = new SimpleDateFormat("yyyyMMdd", Locale.ENGLISH);
+        DateFormat dateTimeFormat = new SimpleDateFormat("yyyyMMddHHmmss", Locale.ENGLISH);
+
+        // MSH Segment
+        MSH msh = message.getMSH();
+        msh.getFieldSeparator().setValue("|");
+        msh.getEncodingCharacters().setValue("^~\\&");
+        msh.getSendingApplication().getNamespaceID().setValue("OpenMRS");
+        msh.getSendingFacility().getNamespaceID().setValue(getOrderLocation(order).getName());
+        // Hard-coding PACS_APP and PACS_FACILITY per instruction from vendor
+        msh.getReceivingApplication().getNamespaceID().setValue("PACS_APP");
+        msh.getReceivingFacility().getNamespaceID().setValue("PACS_FACILITY");
+        msh.getDateTimeOfMessage().getTimeOfAnEvent().setValue(dateTimeFormat.format(now));
+        msh.getMessageType().getMessageType().setValue("ORM");
+        msh.getMessageType().getTriggerEvent().setValue("O01");
+        msh.getMessageControlID().setValue(UUID.randomUUID().toString());
+        msh.getProcessingID().getProcessingID().setValue("P"); // P=Production, D=Debugging, T=Testing
+        msh.getVersionID().setValue("2.3"); // HL7 version targeted
+
+        // PID Segment
+        Patient patient = order.getPatient();
+        PID pid = message.getPATIENT().getPID();
+        pid.getSetIDPatientID().setValue("1");
+        pid.getPatientIDInternalID(0).getID().setValue(patient.getUuid());
+        // Patient Name has a maximum length of 64 in the RIS.  Trim first and last to 30 to ensure it fits.
+        pid.getPatientName(0).getFamilyName().setValue(trim(patient.getFamilyName(), 30));
+        pid.getPatientName(0).getGivenName().setValue(trim(patient.getGivenName(), 30));
+        if (patient.getBirthdate() != null) {
+            pid.getDateOfBirth().getTimeOfAnEvent().setValue(dateFormat.format(order.getPatient().getBirthdate()));
+        }
+        pid.getSex().setValue(patient.getGender() == null ? "O" : patient.getGender().toUpperCase());
+
+        // TODO: Note, these seems like the wrong fields, but it is what is in the integration docs
```

Comment:
Yeah, back in the day, McKesson was very loose about where things go, so not surprising that a vendor might pick the wrong field for someting.